src/gsi-ncdiag/gsi_ncdiag.py

```diff
@@ -1314,3 +1332,211 @@ def toIODAobs(self, OutDir, clobber=True):
+
+        # set up output file
+        ncout = nc.Dataset(outname, 'w', format='NETCDF4')
+        ncout.setncattr("date_time", np.int32(self.validtime.strftime("%Y%m%d%H")))
+        # get nlocs
+        nlocs = self.nobs
+        ncout.createDimension("nlocs", nlocs)
+        # other dims
+        ncout.createDimension("nlevs", self.df.dimensions["nlevs"].size)
+        #ncout.createDimension("nlevsp1", self.df.dimensions["air_pressure_levels_arr_dim"].size)
+        for var in self.df.variables.values():
+            vname = var.name
+            if vname in geovals_metadata_dict.keys():
+                dims = ("nlocs",)
+                var_out = ncout.createVariable(geovals_metadata_dict[vname], var.dtype, dims)
+                vdata = var[:]
+                var_out[:] = vdata
+            elif vname in geovals_vars.keys():
+                if (len(var.dimensions) == 1):
+                    dims = ("nlocs",)
+                elif "_levels" in vname:
+                    dims = ("nlocs", "nlevsp1")
+                else:
+                    dims = ("nlocs", "nlevs")
+                var_out = ncout.createVariable(geovals_vars[vname], var.dtype, dims)
+                vdata = var[...]
+                var_out[...] = vdata
+            else:
+                pass
+        ncout.close()
+
+    def toIODAobs(self, OutDir, clobber=True):
+        """ toIODAobs(OutDir,clobber=True)
+   output observations from the specified GSI diag file
+   to the JEDI/IODA observation format
+        """
+        import ioda_conv_ncio as iconv
+        import netCDF4 as nc
+        import os
+        from collections import defaultdict, OrderedDict
+        from orddicts import DefaultOrderedDict
+        import numpy as np
+        import datetime as dt
+        # set up a NcWriter class
+        outname = OutDir+'/'+self.sensor+'_'+self.obstype+'_obs_'+self.validtime.strftime("%Y%m%d%H")+'.nc4'
+        if not clobber:
+            if (os.path.exists(outname)):
+                print("File exists. Skipping and not overwriting:")
+                print(outname)
+                return
+        RecKeyList = []
+        LocKeyList = []
+        LocVars = []
+        AttrData = {}
+        varDict = defaultdict(lambda: defaultdict(dict))
+        outdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        rec_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        loc_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        var_mdata = defaultdict(lambda: DefaultOrderedDict(OrderedDict))
+        # get list of location variable for this var/platform
+        for ncv in self.df.variables:
+            if ncv in all_LocKeyList:
+                LocKeyList.append(all_LocKeyList[ncv])
+                LocVars.append(ncv)
+        # for now, record len is 1 and the list is empty?
+        recKey = 0
+        writer = iconv.NcWriter(outname, RecKeyList, LocKeyList)
+
+        nlocs = self.nobs
+        print self.obstype
```

Comment:
Please change to python 3 syntax. Thanks!

Reply:
Oops, I'll just delete that print statement.